libs/feature/admin/src/lib/user/edit-user-dialog.tsx

```diff
@@ -0,0 +1,263 @@
+	const height = window.innerHeight * 0.6;
+
+	const isValidDate = (date: Date | null | undefined) => {
+		if (!date) return false;
+		if (date === null) return false;
+		return date instanceof Date && !isNaN(date.getTime());
+	};
+
+	return (
+		<div className="overflow-y-auto overflow-x-hidden p-2" style={{ maxHeight: height }}>
+			<h2 className="mb-4 text-2xl">Daten</h2>
+			<div className="flex flex-col gap-4">
+				<LabeledField label="Name" error={errors.displayName?.message}>
+					<input className="textfield" type={"text"} {...register("name")} />
+				</LabeledField>
+				<LabeledField label="Displayname" error={errors.displayName?.message}>
+					<input className="textfield" type={"text"} {...register("displayName")} />
+				</LabeledField>
+				<LabeledField label="Email" error={errors.email?.message}>
+					<input className="textfield" type={"text"} {...register("email")} />
+				</LabeledField>
+				<LabeledField label="Role" error={errors.role?.message}>
+					<select
+						value={watch("role")}
+						onChange={v => {
+							setValue(
+								"role",
+								v.target.value as UserRole
+							); /*TODO Ask Marcel about this*/
+						}}
+						className="textfield w-64 rounded-lg px-8"
+					>
+						{Object.values(UserRole).map(role => (
+							<option key={"option:" + role} value={role}>
+								{role}
+							</option>
+						))}
+					</select>
+				</LabeledField>
+				<LabeledField label="Email Verified Date" error={errors.emailVerified?.message}>
+					<input
+						className="textfield"
+						type={"date"}
+						value={
+							isValidDate(watch("emailVerified"))
+								? watch("emailVerified")?.toISOString().slice(0, 10)
+								: ""
+						}
+						onChange={v => {
+							setValue("emailVerified", new Date(v.target.value));
+						}}
+					/>
+				</LabeledField>
+				<LabeledField label="Bild" error={errors.image?.message}>
+					<div className="flex w-full gap-4">
+						<div className="flex w-full flex-col gap-2">
+							<input
+								className="textfield w-full"
+								type={"text"}
+								placeholder={"https://example.com/image.png"}
+								{...register("image", { required: false })}
+							/>
+							<Upload
+								mediaType="image"
+								onUploadCompleted={url => setValue("image", url)}
+								preview={
+									<>
+										{imgUrl && imgUrl.length > 0 ? (
+											<img
+												src={imgUrl}
+												alt="Avatar"
+												className="mx-auto h-32 w-32 shrink-0 rounded-lg object-cover"
+											></img>
+										) : (
+											<div className="mx-auto h-32 w-32  shrink-0 rounded-lg bg-gray-200"></div>
+										)}
+									</>
+								}
+							/>
+						</div>
+					</div>
+				</LabeledField>
+			</div>
+		</div>
+	);
+}
+
+function ActionButtons({ username }: { username: string }) {
+	const { mutateAsync: deleteUserAndDependentData } =
+		trpc.admin.deleteUserAndDependentData.useMutation();
+	const [deleteData, setDeleteData] = useState(false);
```

Comment:
ist das eine art mutex um den sync zu behalten? das wäre schon wirklich wirklich unschön. wenn sich das verhalten der komponente anders ist solange das läuft, dann doch lieber await nutzen

Alternativ: wenn es nur um den status indicator geht, einfach `isLoading`verwenden. Den State in der form sollte aber entfernt werden


```suggestion
	const { mutateAsync: deleteUserAndDependentData, isLoading } =
		trpc.admin.deleteUserAndDependentData.useMutation();
    // ...
    { isLoading ? "Deleting" : "Deleted"} 
```